src/model/Model.cpp

```diff
@@ -742,12 +1280,393 @@ namespace model {
+            }
+          } else {
+            workspaceObjects.push_back(componentObject);
+          }
+        } else if (componentObject.optionalCast<ZoneCapacitanceMultiplierResearchSpecial>()) {
+          if (zoneCapacitanceMultiplierResearchSpecial) {
+            for (unsigned i = 1, n = componentObject.numFields(); i < n; ++i) {
+              zoneCapacitanceMultiplierResearchSpecial->setString(i, componentObject.getString(i).get());
+            }
+          } else {
+            workspaceObjects.push_back(componentObject);
+          }
+        } else if (componentObject.optionalCast<ConvergenceLimits>()) {
+          if (convergenceLimits) {
+            for (unsigned i = 1, n = componentObject.numFields(); i < n; ++i) {
+              convergenceLimits->setString(i, componentObject.getString(i).get());
+            }
+          } else {
+            workspaceObjects.push_back(componentObject);
+          }
+        } else if (componentObject.optionalCast<ShadowCalculation>()) {
+          if (shadowCalculation) {
+            for (unsigned i = 1, n = componentObject.numFields(); i < n; ++i) {
+              shadowCalculation->setString(i, componentObject.getString(i).get());
+            }
+          } else {
+            workspaceObjects.push_back(componentObject);
+          }
+        } else if (componentObject.optionalCast<Timestep>()) {
+          if (timestep) {
+            for (unsigned i = 1, n = componentObject.numFields(); i < n; ++i) {
+              timestep->setString(i, componentObject.getString(i).get());
+            }
+          } else {
+            workspaceObjects.push_back(componentObject);
+          }
+        } else if (componentObject.optionalCast<ClimateZones>()) {
+          if (climateZones) {
+            for (unsigned i = 1, n = componentObject.numFields(); i < n; ++i) {
+              climateZones->setString(i, componentObject.getString(i).get());
+            }
+          } else {
+            workspaceObjects.push_back(componentObject);
+          }
+        } else if (componentObject.optionalCast<EnvironmentalImpactFactors>()) {
+          if (environmentalImpactFactors) {
+            for (unsigned i = 1, n = componentObject.numFields(); i < n; ++i) {
+              environmentalImpactFactors->setString(i, componentObject.getString(i).get());
+            }
+          } else {
+            workspaceObjects.push_back(componentObject);
+          }
+        } else if (componentObject.optionalCast<ExternalInterface>()) {
+          if (externalInterface) {
+            for (unsigned i = 1, n = componentObject.numFields(); i < n; ++i) {
+              externalInterface->setString(i, componentObject.getString(i).get());
+            }
+          } else {
+            workspaceObjects.push_back(componentObject);
+          }
+        } else {
+          // not a unique model object
+          workspaceObjects.push_back(componentObject);
+        }
+      }
```

Comment:
Replace everything with this:

```suggestion
      // Start by deleting existing Unique ModelObject, without touching children/resources by calling WorkspaceObject::remove
      for (const WorkspaceObject& componentObject : component.objects()) {
        auto compIddObject = componentObject.iddObject();
        if (compIddObject.properties().unique) {
          // Remove any existing objects (there should really be only one)
          for (WorkspaceObject& wo : model().getObjectsByType(compIddObject)) {
            LOG(Info, "Removing existing UniqueModelObject in the target model: " << wo.briefDescription());
            wo.remove();
          }
        }
      }
```